### PR TITLE
enable gosec and fix related errors

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: gy95 <1015105054@qq.com>

enable gosec linter check
the error output is ` G601: Implicit memory aliasing in for loop. (gosec)`
in the `for range` loop, we should address the value directly, maybe the address points to the same address

ref: [using-reference-to-loop-iterator-variable](https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable)

### CODE REVIEW

File: pkg/scheduler/plugins/numaaware/numaaware.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ func (pp *numaPlugin) OnSessionOpen(ssn *framework.Session) {
 
 			klog.V(4).Infof("[numaaware] hits for task %s container '%v': %v on node %s, besthit: %v",
 				task.Name, container.Name, providersHints, node.Name, hit)
-			resAssignMap := policy.Allocate(&container, &hit, node.NumaSchedulerInfo, resNumaSets, pp.hintProviders)
+			resAssignMap := policy.Allocate(&con, &hit, node.NumaSchedulerInfo, resNumaSets, pp.hintProviders)
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/numaaware/numaaware.go:140]: So why new thest variables?

Comment (gy95) [reply pkg/scheduler/plugins/numaaware/numaaware.go:140]: for-range is not recommended reference value directly, so new variable to fix lint checks.

**Comment by jasonliu747** [comment]
Seems a little bit not necessary to me, but...
image

**Review by jasonliu747 — COMMENTED** [review]
Seems a little bit not necessary to me, but...
image

File: pkg/scheduler/plugins/numaaware/numaaware.go (hunk 1 above)
Comment (Thor-wl) [reply pkg/scheduler/plugins/numaaware/numaaware.go:140]: Well, thanks for the advice. Maybe we can pick some other bug fix which is more valuable next time.

File: example/kubecon-2019-china/scripts/node-info.go [hunk 2]
```diff
@@ -61,7 +61,8 @@ func main() {
 		if _, found := podMap[nodeName]; !found {
 			podMap[nodeName] = api.EmptyResource()
 		}
-		res := api.GetPodResourceWithoutInitContainers(&p)
+		pod := p
```
Comment (k82cn) [inline example/kubecon-2019-china/scripts/node-info.go:64]: honestly, I don-t think that's necessary fix for this lint :(

**Review by hzxuzhonghu — COMMENTED** [review]
Overall, i donot think it is worth  to introduce gosec

**Pull request closed without merging**